### PR TITLE
add username modal

### PR DESCRIPTION
**Pull request by stevekrouse**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "keywords": [],
   "main": "src/index.js",
   "dependencies": {
-    "@compose-run/client": "^0.0.17",
+    "@compose-run/client": "^0.0.18",
```
Comment (stevekrouse) [inline package.json]: This was to fix some bugs I found with `useCloudReducer`

File: src/components/LoginModal.tsx [hunk 2]
```diff
@@ -58,11 +58,13 @@ export default function LoginModal({
...
                 marginTop: "25px",
                 paddingLeft: ".3em",
+                // would be nice to have it show the input border if the input is not selected
+                // but there's no css hover state selectors natively in react
+                // maybe we'll move to tailwind or some other lib that supports that
                 border: "none",
               }}
               placeholder="you@email.com"
               onKeyPress={async (e) => {
-                // TODO - disable for mobile
```
Comment (stevekrouse) [inline src/components/LoginModal.tsx:65]: I've found that at least on my google pixel, this is handled well on phones already 🤷

File: src/state/messages.ts [hunk 3]
```diff
@@ -87,4 +87,4 @@ export const useMessages = () =>
       }
       return messages;
     },
-  }) as [MessagesDB, (action: MessageAction) => Promise<MessageActionError>];
```
Comment (stevekrouse) [inline src/state/messages.ts:90]: Don't need these annotations now that I fixed `useCloudReducer`'s type

File: src/state/users.ts [hunk 4]
```diff
@@ -6,22 +6,14 @@ type UsersDB = { [userId: number]: string };
 export type UserAction = string;
 type UserActionError = string;
 
-// interface SetUserName {
-//   type: "SetUserName";
-//   username: string;
-// }
-
-const usersName = `${appName}/user-settings`;
+const usersName = `${appName}/user-settings-1`;
```
Comment (stevekrouse) [inline src/state/users.ts:9]: I had to change the name so that I could edit the reducer. It's a super weird (bad) convention, but it kinda works for now...

**Comment by stevekrouse** [comment]
I think this branch is ready to merge with main, but I'll leave it open for @parenthetical to see the diff with #3. When you're ready @parenthetical, we can change the base branch and merge it to main 👍

File: src/state/users.ts [hunk 5]
```diff
@@ -6,22 +6,14 @@ type UsersDB = { [userId: number]: string };
 export type UserAction = string;
```
Comment (stevekrouse) [inline src/state/users.ts:6]: We will probably soon want:

```ts
type UserId = number;
type UsersDB = { [userId: number]: User };
export type UserAction = SetUsername | ...
export interface User = { id: UserId, username: string, ...}
```

But we can leave this for another PR

Comment (parenthetical) [reply src/state/users.ts:6]: Totally, I thought I'd just do exactly what we need and have the type system help us refactor later :)
